IdentityCore/src/webview/embeddedWebview/challangeHandlers/MSIDPKeyAuthHandler.m

```diff
@@ -72,6 +73,23 @@ + (BOOL)handleChallenge:(NSString *)challengeUrl
     NSMutableURLRequest *responseReq = [[NSMutableURLRequest alloc] initWithURL:responseUrlComp.URL];
     [responseReq setValue:kMSIDPKeyAuthHeaderVersion forHTTPHeaderField:kMSIDPKeyAuthHeader];
     [responseReq setValue:authHeader forHTTPHeaderField:MSID_OAUTH2_AUTHORIZATION];
+
+    // Adding refreshTokenCredential (PRT) header to the challenge response. Header is available in customheaders dict
+    BOOL refreshTokenIsValid = NO;
+    if (customHeaders != nil && customHeaders.count > 0)
+    {
+        NSString *primaryRefreshToken = [customHeaders objectForKey:MSID_REFRESH_TOKEN_CREDENTIAL];
+        if (primaryRefreshToken != nil)
+        {
+            [responseReq setValue: primaryRefreshToken forHTTPHeaderField:MSID_REFRESH_TOKEN_CREDENTIAL];
```

Comment:
nit: Let's add a LogInfo level message like "Added refresh token to the PkeyAuth response".